client/influxdb.go

```diff
@@ -183,6 +239,52 @@ func (c *Client) Write(bp BatchPoints) (*Response, error) {
 	return nil, nil
 }
 
+// WriteLineProtocol takes a string with line returns to delimit each write
+// If successful, error is nil and Response is nil
+// If an error occurs, Response may contain additional information if populated.
+func (c *Client) WriteLineProtocol(data, database, retentionPolicy, precision, writeConsistency string) (*Response, error) {
+	c.url.Path = "write"
+
+	var b bytes.Buffer
+	b.WriteString(data)
+
+	req, err := http.NewRequest("POST", c.url.String(), &b)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "")
+	req.Header.Set("User-Agent", c.userAgent)
+	if c.username != "" {
+		req.SetBasicAuth(c.username, c.password)
+	}
+	params := req.URL.Query()
+	params.Add("db", database)
+	params.Add("rp", retentionPolicy)
+	params.Add("precision", precision)
+	params.Add("consistency", writeConsistency)
+	req.URL.RawQuery = params.Encode()
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response Response
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil && err.Error() != "EOF" {
+		return nil, err
+	}
```

Comment:
`ioutil.ReadAll()` doesn't return an EOF. Also, if you're checking for EOF you can compare by reference: `err != io.EOF`.